src/app/services/xml-parsers/generic-parser.service.ts

```diff
@@ -1,171 +1,8 @@
-      type: GenericElementData,
-      class: xml.tagName ? xml.tagName.toLowerCase() : '',
-      content: this.parseChildren(xml),
-      attributes: this.parseAttributes(xml),
-    };
-
-    return genericElement;
-  }
-
-  private parseNote(xml: XMLElement): NoteData {
-    let noteLayout: NoteLayout = 'popover';
-    if (isNestedInElem(xml, 'div', [{ key: 'type', value: 'footer' }]) // FOOTER NOTE
-      || isNestedInElem(xml, 'person') || isNestedInElem(xml, 'place') || isNestedInElem(xml, 'org')
-      || isNestedInElem(xml, 'relation') || isNestedInElem(xml, 'event') // NAMED ENTITY NOTE
-    ) {
-      noteLayout = 'plain-text';
-    }
-    let noteType = xml.getAttribute('type') ?? 'comment';
-    if (isNestedInElem(xml, 'app')) {
-      noteType = 'critical';
-    }
-
-    const noteElement = {
-      type: NoteData,
-      noteType,
-      noteLayout,
-      exponent: xml.getAttribute('n'),
-      path: xpath(xml),
-      content: this.parseChildren(xml),
-      attributes: this.parseAttributes(xml),
-    };
-
-    return noteElement;
-  }
-
-  private parseNamedEntityRef(xml: XMLElement): NamedEntityRefData | GenericElementData {
-    const ref = xml.getAttribute('ref');
-    if (!ref) {
-      return this.parseElement(xml);
-    }
-
-    const neTypeMap: { [key: string]: NamedEntityType } = {
-      placename: 'place',
-      geogname: 'place',
-      persname: 'person',
-      orgname: 'org',
-      event: 'event',
-    };
-
-    return {
-      type: NamedEntityRefData,
-      entityId: ref ? ref.replace(/#/g, '') : '',
-      entityType: neTypeMap[xml.tagName.toLowerCase()],
-      path: xpath(xml),
-      content: this.parseChildren(xml),
-      attributes: this.parseAttributes(xml),
-      class: xml.tagName.toLowerCase(),
-    };
-  }
-
-  private parsePtr(xml: XMLElement) {
-    if (xml.getAttribute('type') === 'noteAnchor' && xml.getAttribute('target')) {
-      const noteId = xml.getAttribute('target').replace('#', '');
-      const rootNode = xml.closest('TEI');
-      const noteEl = rootNode.querySelector<XMLElement>(`note[*|id="${noteId}"]`);
-
-      return noteEl ? this.parseNote(noteEl) : this.parseElement(xml);
-    }
-
-    return this.parseElement(xml);
-  }
-
-  private parseLb(xml: XMLElement) {
-    return {
-      id: xml.getAttribute('xml:id') || xpath(xml),
-      n: xml.getAttribute('n') || '',
-      rend: xml.getAttribute('rend'),
-      facs: xml.getAttribute('facs'),
-      type: LbData,
-      content: [],
-      attributes: this.parseAttributes(xml),
-    };
-  }
-
-  private parseChildren(xml: XMLElement) {
-    return complexElements(xml.childNodes).map(child => this.parse(child as XMLElement));
-  }
-
-  public parseAttributes(xml: XMLElement): AttributesData {
-    return Array.from(xml.attributes)
-      .map(({ name, value }) => ({ [name === 'xml:id' ? 'id' : name.replace(':', '-')]: value }))
-      .reduce((x, y) => ({ ...x, ...y }), {});
-  }
+  parse(xml: XMLElement) { return parse(xml); }
```

Comment:
Why don't just delete `GenericParserService` since it only has a method that is simply a wrapper to `parse` function?

Reply:
Cause we will add other functionalities, like loading state.